libs/utils/src/postgres_connection.rs

```diff
@@ -0,0 +1,162 @@
+use anyhow::Context;
+use std::fmt;
+use url::Host;
+
+pub fn parse_host_port<S: AsRef<str>>(host_port: S) -> Result<(Host, Option<u16>), anyhow::Error> {
```

Comment:
There is a similar function [in proxy](https://github.com/neondatabase/neon/blob/f564dff0e320a859ab9eb8b40caa9a70599c829a/proxy/src/auth/backend/console.rs#L252-L255), but I wouldn't touch it in this PR, so there's less to review. Maybe a separate PR, one can argue that IPv6 support is good.

Reply:
You might want to use [`SocketAddr`](https://doc.rust-lang.org/std/net/enum.SocketAddr.html) or [`ToSocketAddrs`](https://doc.rust-lang.org/std/net/trait.ToSocketAddrs.html) here instead.

In fact, I'll likely use one of those in the proxy.

Reply:
`Host` also includes unresolved hostnames. I suspect it's more obvious to resolve on connect rather than on configuration parsing. So I'd keep it as it is.